dlt/pipeline/trace.py

```diff
@@ -162,22 +166,24 @@ def on_end_trace(self, trace: PipelineTrace, pipeline: SupportsPipeline) -> None
         ...
 
 
-# plug in your own tracking module here
-# TODO: that probably should be a list of modules / classes with all of them called
-TRACKING_MODULE: SupportsTracking = None
+# plug in your own tracking modules here
+TRACKING_MODULES: SupportsTracking = None
 
 
 def start_trace(step: TPipelineStep, pipeline: SupportsPipeline) -> PipelineTrace:
     trace = PipelineTrace(uniq_id(), pendulum.now(), steps=[])
-    with suppress_and_warn():
-        TRACKING_MODULE.on_start_trace(trace, step, pipeline)
+    for module in TRACKING_MODULES:
+        with suppress_and_warn():
+            module.on_start_trace(trace, step, pipeline)
     return trace
 
 
 def start_trace_step(trace: PipelineTrace, step: TPipelineStep, pipeline: SupportsPipeline) -> PipelineStepTrace:
     trace_step = PipelineStepTrace(uniq_id(), step, pendulum.now())
-    with suppress_and_warn():
-        TRACKING_MODULE.on_start_trace_step(trace, step, pipeline)
+    trace.steps.append(trace_step)
```

Comment:
is there any downside to attaching the trace step on trace start? it would be nicer to have it like this when sent to the platform